hr_prometheus/utils.py

```diff
@@ -1,8 +1,23 @@
+from contextlib import ContextDecorator
 from enum import Enum
+from time import time
 
 
-def apply_labels(metric, labels):
-    resolved_labels = [
-        label.value if isinstance(label, Enum) else label for label in labels
-    ]
-    return metric.labels(*resolved_labels)
+class TimeMonitor(ContextDecorator):
+    def __init__(self, metric, labels):
+        self.metric = metric
+        self.labels = labels
```

Comment:
Nice change, but given the fact that we are already using an object, why don't we store the resolved metric instead of the general one and the labels? e.g.:
```python
class TimeMonitor(ContextDecorator):
    def __init__(self, metric, labels):
        self.labeled_metric = apply_labels(metric, labels)
```
And leave `apply_labels` as a pure function?

Reply:
I like, nicer solution.